spec/controllers/shift_spec.rb

```diff
@@ -0,0 +1,208 @@
+require "rails_helper"
+
+RSpec.describe "Shifts", type: :request do
+  describe "GET /index" do
+    before do
+      FactoryBot.create_list(:shift, 10)
+      @shift = Shift.first
+      get "/api/v1/shifts"
+      @first_shift_data = json["data"][0]["attributes"]
+    end
+    
+    it "returns status code 200 (success)" do
+      expect(response).to have_http_status(:success)
+    end
+
+    it "returns all shifts" do
+      expect(json["data"].size).to eq(10)
+    end
+
+    it "returns correct shift_start" do
+      shift_start_year = @first_shift_data["shift_start"].split("T").first.split("-").first.to_i
+      shift_start_month = @first_shift_data["shift_start"].split("T").first.split("-").second.to_i
+      shift_start_day = @first_shift_data["shift_start"].split("T").first.split("-").third.to_i
+      shift_start_hour = @first_shift_data["shift_start"].split("T").second.split(".").first.split(":").first.to_i
+      shift_start_minute = @first_shift_data["shift_start"].split("T").second.split(".").first.split(":").second.to_i
+      shift_start_second = @first_shift_data["shift_start"].split("T").second.split(".").first.split(":").third.to_i
+      json_shift_start = Time.new(shift_start_year, shift_start_month, shift_start_day, shift_start_hour, shift_start_minute, shift_start_second)
+
+      @shift_start_year = @shift.shift_start.to_s.split(" ").first.split("-").first.to_i
+      @shift_start_month = @shift.shift_start.to_s.split(" ").first.split("-").second.to_i
+      @shift_start_day = @shift.shift_start.to_s.split(" ").first.split("-").third.to_i
+      @shift_start_hour = @shift.shift_start.to_s.split(" ").second.split(":").first.to_i
+      @shift_start_minute = @shift.shift_start.to_s.split(" ").second.split(":").second.to_i
+      @shift_start_second = @shift.shift_start.to_s.split(" ").second.split(":").third.to_i
```

Comment:
Ok, two major things here:

1. You can take the string of a Time object and parse it back into a Time object.  `Time.parse(@first_shift_data["shift_start"])`.  Manually parsing is error-prone
2. You can get the year/month/day/hour/min/sec by calling methods by those names instead of converting to a string and doing complex string parsing.  This will even get you integers.
3. What is the benefit of taking a Time object, splitting it into its component parts, and then re-assembling a Time object?  If it's an incompatible Time object type (DateTime, Date, Time) to another Time object, they all have `to_TYPE` methods to convert to each other for comparison.

I'm only gonna comment in this one spot, but you appear to have replicated this pattern a bunch in the tests.